packages/vscode-apollo/README.md

```diff
@@ -1,18 +1,154 @@
+
+1. Create an `apollo.config.js` at the root of the project
+2. Copy an API key from the Engine dashboard of the published service
+
+<h3 id="apollo-config">Setting up an Apollo config</h3>
+In order for the VS Code plugin to know how to find the schema, it needs to be linked to either a published schema or a local one. To link a project to a published schema, edit the `apollo.config.js` file to look like this:
+
+```js
+module.exports = {
+  client: {
+    service: "my-graphql-app"
+  }
+};
+```
+
+The service name is the id of the service created in Engine and can be found in the services dashboard of [Engine](https://engine.apollographql.com)
+
+> Important note: If the name of the service in Engine is changed, this value should be the service id. This can be found in the url when browsing the service in Engine. This will be easier to manage in the near future
+
+<h3 id="api-key">Setting up an API key</h3>
+To authenticate with Engine to pull down the schema, create a file next to the `apollo.config.js` called `.env`. This should be an untraced file (i.e. don't push it to GitHub). Go to the settings page of the published service and create a new API key.
+
+> It is best practice to create a new API key for each member of the team and name the key so its easy to find and revoke if needed
+
+After the key is found, add the following line to the `.env` file:
+
+```bash
+ENGINE_API_KEY=<enter copied key here>
+```
+
+After this is done, VS Code can be restarted and the editor integration will start providing autocomplete, validation, and more!
+
+<h3 id="local-schemas">Local schemas</h3>
+
+Sometimes it may make sense to link the editor to a locally running version of a schema to try out new designs that are in active development. To do this, the `apollo.config.js` file can be linked to a local service definition:
+
+```js
+module.exports = {
+  client: {
+    service: {
+      name: "my-graphql-app",
+      url: "http://localhost:4000/graphql"
+    }
+  }
+};
+```
+
+> Linking to the local schema won't provide all features such as switching schema tags and performance metrics.
+
+More information about configuring an Apollo project can be found [here](../platform/apollo-config.html)
+
+<h3 id="client-only-schemas">Client-only schemas</h3>
+
+One of the best features of the VS Code extension is the automatic merging of remote schemas and local ones when using integrated state management with Apollo Client. This happens automatically whenever schema definitions are found within a client project.
+
+Client side schema definitions can be spread throughout the client app project and will be merged together to create one single schema. This can be controlled through the `apollo.config.js` at the root of he project:
```

Comment:
```suggestion
Client-side schema definitions can be spread throughout the client app project and will be merged together to create one single schema. This can be controlled through the `apollo.config.js` at the root of the project:
```